pandas/core/generic.py

```diff
@@ -3210,7 +3210,7 @@ def _maybe_cache_changed(self, item, value) -> None:
         """
         The object has called back to us saying maybe it has changed.
         """
-        self._mgr.set(item, value)
+        NDFrame._set_item(self, item, value, clear=False)
```

Comment:
for my edification, why NDFrame?

Reply:
because we dont want to go through the DataFrame._set_item method

Reply:
are double underscore methods appropriate here for intraclass method calls? https://docs.python.org/3/tutorial/classes.html?highlight=mangle#private-variables